.changeset/angry-days-swim.md

```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-frontend/application-shell-connectors': patch
+---
+
+Create a new React hook to access Projects Images rewrite configuration. At the moment, most components are not functional, moving away from HOC, we want to have a react hook that can get context.
```

Comment:
I would suggest to update the changeset comment to be something like this:

---

There is a new React hook which you can use to access the Project Images rewrite configuration ([see documentation](https://docs.commercetools.com/custom-applications/api-reference/commercetools-frontend-application-shell-connectors#project-image-settings)).

You would use it like this:
```
function MyComponent() {
  const { isLoading, imageRegex } = useProjectExtensionImageRegex();

  if (isLoading) return <LoadingSpinner />;

  return (
    <div>
      <h1>Project images regex: {imageRegex}</h1>
    </div>
  );
}

function MyApp() {
  return (
    <ProjectExtensionProviderForImageRegex>
      <MyComponent />
    </ProjectExtensionProviderForImageRegex>
  );
}
```

Both `GetProjectExtensionImageRegex` component and `withProjectExtensionImageRegex` still exists for backwards compatibility but have been marked as deprecated.